components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -381,8 +385,24 @@ protected void sendResponse(HttpServletRequest request, HttpServletResponse resp
         }
 
         // redirect to the caller
-        String redirectURL = context.getCallerPath() + "?sessionDataKey="
-                             + context.getCallerSessionKey() + rememberMeParam;
+        String webContextRoot = ServerConfiguration.getInstance().getFirstProperty(FrameworkConstants.WEB_CONTEXT_ROOT);
+        if (StringUtils.isNotBlank(webContextRoot)) {
+            if (webContextRoot.charAt(0) == '/') {
+                if (webContextRoot.length() <= 1) {
+                    webContextRoot = "";
+                }
+            } else {
+                webContextRoot = "/" + webContextRoot;
+            }
+        }
+        String redirectURL;
+        if (StringUtils.isNotBlank(webContextRoot)) {
+            redirectURL = webContextRoot + context.getCallerPath() + "?sessionDataKey="
+                    + context.getCallerSessionKey() + rememberMeParam;
+        } else {
+            redirectURL = context.getCallerPath() + "?sessionDataKey="
+                    + context.getCallerSessionKey() + rememberMeParam;
+        }
```

Comment:
Need to find out why we need these changes in authentication framework. Ideally they should all be part of the caller path variable send in the request.